<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SlackTreesDialogBase</class>
 <widget class="QDialog" name="SlackTreesDialogBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>454</width>
    <height>240</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SlackTrees</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0" colspan="2">
    <widget class="QLabel" name="InputLabel">
     <property name="text">
      <string>Input point layer</string>
     </property>
    </widget>
   </item>
   <item row="0" column="2" colspan="7">
    <widget class="QgsMapLayerComboBox" name="InputMapLayerComboBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QComboBox::AdjustToContents</enum>
     </property>
     <property name="filters">
      <set>QgsMapLayerProxyModel::HasGeometry|QgsMapLayerProxyModel::LineLayer|QgsMapLayerProxyModel::NoGeometry|QgsMapLayerProxyModel::PluginLayer|QgsMapLayerProxyModel::PointLayer|QgsMapLayerProxyModel::PolygonLayer|QgsMapLayerProxyModel::VectorLayer</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="SpacingLabel">
     <property name="text">
      <string>Spacing</string>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QgsDoubleSpinBox" name="SpacingDoubleSpinBox">
     <property name="decimals">
      <number>1</number>
     </property>
     <property name="maximum">
      <double>50.000000000000000</double>
     </property>
     <property name="value">
      <double>2.000000000000000</double>
     </property>
     <property name="showClearButton">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="LengthLabel">
     <property name="text">
      <string>Length</string>
     </property>
    </widget>
   </item>
   <item row="4" column="2" colspan="4">
    <widget class="QLineEdit" name="OutputLIneEdit">
     <property name="text">
      <string/>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="2" colspan="2">
    <widget class="QgsDoubleSpinBox" name="LengthMinDoubleSpinBox">
     <property name="decimals">
      <number>1</number>
     </property>
     <property name="maximum">
      <double>499.000000000000000</double>
     </property>
     <property name="value">
      <double>10.000000000000000</double>
     </property>
     <property name="showClearButton">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="4">
    <widget class="QLabel" name="LengthMinLabel">
     <property name="text">
      <string>min</string>
     </property>
    </widget>
   </item>
   <item row="2" column="5" colspan="3">
    <widget class="QgsDoubleSpinBox" name="LengthMaxDoubleSpinBox">
     <property name="decimals">
      <number>1</number>
     </property>
     <property name="minimum">
      <double>1.000000000000000</double>
     </property>
     <property name="maximum">
      <double>500.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>1.000000000000000</double>
     </property>
     <property name="value">
      <double>50.000000000000000</double>
     </property>
     <property name="showClearButton">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="2" column="8">
    <widget class="QLabel" name="LengthMaxLabel">
     <property name="text">
      <string>max</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="ExcludeLabel">
     <property name="text">
      <string>Exclude</string>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QgsFieldComboBox" name="ExcludeFieldComboBox"/>
   </item>
   <item row="3" column="3" colspan="3">
    <widget class="QComboBox" name="ExcludeComboBox">
     <item>
      <property name="text">
       <string>&gt;</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>&lt;</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>&gt;=</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>&lt;=</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>==</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>!=</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="3" column="6" colspan="3">
    <widget class="QLineEdit" name="ExcludeLineEdit"/>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="OutputLabel">
     <property name="text">
      <string>Output lines</string>
     </property>
    </widget>
   </item>
   <item row="4" column="6" colspan="3">
    <widget class="QPushButton" name="OutputBrowseButton">
     <property name="text">
      <string>Browse</string>
     </property>
    </widget>
   </item>
   <item row="6" column="0" colspan="2">
    <widget class="QCheckBox" name="CanvasCheckBox">
     <property name="text">
      <string>Add to canvas</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QDialogButtonBox" name="HelpButton">
     <property name="standardButtons">
      <set>QDialogButtonBox::Help</set>
     </property>
    </widget>
   </item>
   <item row="7" column="1" colspan="3">
    <widget class="QProgressBar" name="ProgressBar">
     <property name="value">
      <number>0</number>
     </property>
    </widget>
   </item>
   <item row="7" column="4" colspan="3">
    <widget class="QDialogButtonBox" name="CancelButton">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel</set>
     </property>
    </widget>
   </item>
   <item row="7" column="7" colspan="2">
    <widget class="QDialogButtonBox" name="OkButton">
     <property name="standardButtons">
      <set>QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="5" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="2" column="9">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>qgsdoublespinbox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsFieldComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsfieldcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>CancelButton</sender>
   <signal>rejected()</signal>
   <receiver>SlackTreesDialogBase</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>230</y>
    </hint>
    <hint type="destinationlabel">
     <x>521</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
